Update to 1.18.x

LifeIsAParadox opened the pull request:
also works with 1.17.x but need java 17

LifeIsAParadox committed:
Update to 1.18.x

also works with 1.17.x but need java 17

kb-1000 commented:
requiring java 17 at runtime while claiming to be compatible with 1.17.x sounds like a bad idea if you ask me
You definitely do need to _compile_ using Java 17. But you do NOT need to _target_ Java 17 (you can target Java 16 using Java 17, while compiling against a Java 17 library). It's entirely possible to do that, and would cause less user confusion.

LifeIsAParadox commented:
> requiring java 17 at runtime while claiming to be compatible with 1.17.x sounds like a bad idea if you ask me You definitely do need to _compile_ using Java 17. But you do NOT need to _target_ Java 17 (you can target Java 16 using Java 17, while compiling against a Java 17 library). It's entirely possible to do that, and would cause less user confusion.

I did not know that it works like that. I tested it and it works. I am uploading the new version with minimum java requirement of 16.

LifeIsAParadox committed:
changed to java 16

This version is compatible with 1.17.x and 1.18.x with the minimum java requirement of 16.
I have now set sourceCompatibility and targetCompatibility to JavaVersion.VERSION_16 and it.options.release to 16 and at mixin the compatibilityLevel also to JAVA_16

mounderfod reviewed (approved):
LGTM
